libcudacxx/include/cuda/__mdspan/host_device_accessor.h

```diff
@@ -0,0 +1,383 @@
+
+template <typename _Accessor>
+inline constexpr bool is_managed_accessor_v<__managed_accessor<_Accessor>> = true;
+
+template <typename _Tp>
+inline constexpr bool is_host_device_managed_accessor_v =
+  is_host_accessor_v<_Tp> || is_device_accessor_v<_Tp> || is_managed_accessor_v<_Tp>;
+
+/***********************************************************************************************************************
+ * Host Accessor
+ **********************************************************************************************************************/
+
+template <typename _Accessor>
+struct __host_accessor : public _Accessor
+{
+  static_assert(!is_host_device_managed_accessor_v<_Accessor>,
+                "cuda::__host_accessor/cuda::__device_accessor/cuda::__managed_accessor cannot be nested");
+
+  using offset_policy    = __host_accessor<typename _Accessor::offset_policy>;
+  using data_handle_type = typename _Accessor::data_handle_type;
+  using reference        = typename _Accessor::reference;
+  using element_type     = typename _Accessor::element_type;
+
+private:
+  static constexpr bool __is_access_noexcept = noexcept(_Accessor{}.access(data_handle_type{}, 0));
+  static constexpr bool __is_offset_noexcept = noexcept(_Accessor{}.offset(data_handle_type{}, 0));
+
+  template <typename data_handle_type>
+  _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool __is_host_accessible_pointer(data_handle_type __p) noexcept
+  {
+    if constexpr (_CUDA_VSTD::is_pointer_v<data_handle_type>)
+    {
+      ::cudaPointerAttributes __attrib{};
+      _CCCL_VERIFY(::cudaPointerGetAttributes(&__attrib, __p) == ::cudaSuccess, "cudaPointerGetAttributes failed");
+      return __attrib.hostPointer != nullptr || __attrib.type == ::cudaMemoryTypeUnregistered;
+    }
+    else
+    {
+      return true; // cannot be verified
+    }
+  }
+
+  _LIBCUDACXX_HIDE_FROM_ABI static constexpr void __check_host_pointer(data_handle_type __p) noexcept
+  {
+    _CCCL_ASSERT(__is_host_accessible_pointer(__p), "cuda::__host_accessor data handle is not a HOST pointer");
+  }
+
+public:
+  _CCCL_TEMPLATE(typename _NotUsed = void)
+  _CCCL_REQUIRES(_CCCL_TRAIT(_CUDA_VSTD::is_default_constructible, _Accessor))
+  _LIBCUDACXX_HIDE_FROM_ABI __host_accessor() noexcept(_CUDA_VSTD::is_nothrow_default_constructible_v<_Accessor>)
+      : _Accessor{}
+  {}
+
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr __host_accessor(const _Accessor& __acc) //
+    noexcept(_CUDA_VSTD::is_nothrow_copy_constructible_v<_Accessor>)
+      : _Accessor{__acc}
+  {}
+
+  template <typename _OtherAccessor>
+  __host_accessor(const __device_accessor<_OtherAccessor>&) = delete;
+
+  _CCCL_TEMPLATE(typename _OtherAccessor)
+  _CCCL_REQUIRES(_CCCL_TRAIT(_CUDA_VSTD::is_constructible, _OtherAccessor))
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr
+#if _CCCL_STD_VER >= 2020
+    explicit(!_CUDA_VSTD::is_convertible_v<_OtherAccessor, _Accessor>)
+#endif
```

Comment:
Critical: This will change behavior in a braking way between C++20 and C++17. We *must* duplicate those constructors, once implicit and once explicit. (You need to additionally constrain the implicit constructor with `_CUDA_VSTD::is_convertible_v<_OtherAccessor, _Accessor>`

Applies throughout